### PR TITLE
Feature/date range filter

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this.from = Preconditions.checkNotNull(from);
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * @return the beginning date of the filter
```
Comment (jeffjohnson9046) [inline src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java]: Isn't this actually the end date of the filter?

File: src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    public Instant getTo() {
+        return to;
+    }
+
+    /**
+     * @return the ending date of the filter
```
Comment (jeffjohnson9046) [inline src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java]: Similarly, isn't this the beginning date of the filter?

File: src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    static final String TYPE = "dateRange";
+
+    private final Instant to;
+    private final Instant from;
+
+    public DateRangeFilter(@JsonProperty("property") final String property,
+                           @JsonProperty("to") final Instant to,
+                           @JsonProperty("from") final Instant from) {
+        super(property);
+        this.to = Preconditions.checkNotNull(to);
```
Comment (jeffjohnson9046) [inline src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java]: Should there be checks to make sure `to` isn't less than `from`?

Comment (millerts) [reply src/main/java/org/opentestsystem/ap/imrt/iss/dto/search/DateRangeFilter.java]: I guess the query runs just fine in IMRT so I didn't see value in returning the error.  You just won't get valid results.

**Pull request merged**